packages/ui/src/viewer-table/DataTable/SchemaTable/Columns/AllowedValues.tsx

```diff
@@ -0,0 +1,260 @@
+			});
+		} else if (count.exclusiveMax !== undefined) {
+			restrictionItems.push({
+				prefix: `Fewer than ${count.exclusiveMax}${delimiterText} from:\n`,
+				content: `${codeListDisplay}`,
+			});
+		}
+	}
+};
+
+export const computeAllowedValuesColumn = (
+	restrictions: CellContext<SchemaField, SchemaRestrictions>,
+): restrictionItem[] => {
+	const schemaField: SchemaField = restrictions.row.original;
+	const restrictionsValue: SchemaRestrictions = restrictions.getValue();
+	const restrictionItems: restrictionItem[] = [];
+
+	if (!restrictionsValue || Object.keys(restrictionsValue).length === 0) {
+		restrictionItems.push({ prefix: null, content: 'None' });
+		return restrictionItems;
+	}
+	restrictionItems.push(...handleDependsOn(restrictionsValue));
+
+	if ('regex' in restrictionsValue && restrictionsValue.regex) {
+		const regexValue =
+			Array.isArray(restrictionsValue.regex) ? restrictionsValue.regex.join(', ') : restrictionsValue.regex;
+		restrictionItems.push({
+			prefix: 'Must match pattern: ',
+			content: `${regexValue}\nSee field description for examples.`,
+		});
+	}
+
+	if ('codeList' in restrictionsValue && restrictionsValue.codeList && !('count' in restrictionsValue)) {
+		const codeListDisplay =
+			Array.isArray(restrictionsValue.codeList) ?
+				restrictionsValue.codeList.join(',\n')
+			:	`"${restrictionsValue.codeList}"`;
+		restrictionItems.push({ prefix: 'One of: \n', content: `${codeListDisplay}` });
+	}
+
+	if ('range' in restrictionsValue && restrictionsValue.range) {
+		handleRange(restrictionsValue.range, restrictionItems);
+	}
+
+	if (
+		'codeList' in restrictionsValue &&
+		restrictionsValue.codeList &&
+		'count' in restrictionsValue &&
+		restrictionsValue.count
+	) {
+		handleCodeListsWithCountRestrictions(
+			restrictionsValue.codeList,
+			restrictionsValue.count,
+			restrictionItems,
+			schemaField.isArray || false,
+			schemaField.delimiter ?? ',',
+		);
+	}
+
+	if (schemaField.unique) {
+		restrictionItems.push({ prefix: null, content: 'Must be unique' });
+	}
+
+	if (restrictionItems.length === 0) {
+		restrictionItems.push({ prefix: null, content: 'None' });
+	}
+
+	return restrictionItems;
+};
+
+const handleDependsOn = (schemaRestrictions: SchemaRestrictions): restrictionItem[] => {
+	const restrictionItems: restrictionItem[] = [];
+
+	if (schemaRestrictions && 'compare' in schemaRestrictions && schemaRestrictions.compare) {
+	}
+	if (
+		schemaRestrictions &&
+		'if' in schemaRestrictions &&
+		schemaRestrictions.if &&
+		'conditions' in schemaRestrictions.if &&
+		schemaRestrictions.if.conditions &&
+		Array.isArray(schemaRestrictions.if.conditions)
+	) {
+		const allFields: string[] = [];
+		schemaRestrictions.if.conditions.forEach((condition) => {
+			if (condition.fields !== undefined) {
+				if (Array.isArray(condition.fields)) {
+					allFields.push(...condition.fields);
+				} else if (typeof condition.fields === 'string') {
+					allFields.push(condition.fields);
+				}
+			}
```

Comment:
all of these "if.. then.. push" style pieces of code would be better written as functional map or reduce functions, that way they are returning new object each time.

the main issue with passing around an object as parameter, mutating it in a function, and then returning it - is you very quickly loose track of what and why is mutating it. slightly more overhead with returning cleaner new objects, but it's a better self contained way of doing it.